### DISCUSSION TITLE
正则问题

### DISCUSSION THREAD

**Discussion by LX-sys** [fastapi/fastapi#8477] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional
from fastapi import FastAPI
from fastapi import Query


app = FastAPI()


@app.get("/items/")
async def read_item(q:Optional[str]=Query(None,regex="^[a-z]*")):
    results = ["a","b"]
    if q:
        results.append(q)
    return results
```

### Description

http://127.0.0.1:8000/items/?q=12345

When I access this URL in my browser,12345 can be matched by the re ^[a-z]*

### Operating System

Windows

### FastAPI Version

0.75.2

### Python Version

3.9

**Comment by arunppsg** [comment, marked as answer]
Maybe you need to use `regex="^[a-z]$"` to match only a-z characters in your query. `^[a-z]*` causes it to match 0 or more repetitions of the preceding regex i.e with input `abc1`, it will only match `ab`.
